pypeit/coadd1d.py

```diff
@@ -207,9 +209,101 @@ def load(self):
+
+        Returns
+        -------
+        gpm_exp: list of bool
+            List of boolean that indicates which exposures
+            have been coadded. The length of the list is nexp.
+        _waves : list of float `numpy.ndarray`_
+            Updated list of wavelength arrays.
+        _fluxes : list of float `numpy.ndarray`_
+            Updated list of flux arrays.
+        _ivars : list of float `numpy.ndarray`_
+            Updated list of inverse variance arrays.
+        _gpms : list of bool `numpy.ndarray`_
+            Updated list of good pixel mask variance arrays.
+        """
+
+        # initialize the exposures lists
+        _waves = [wave for wave in self.waves]
+        _fluxes = [flux for flux in self.fluxes]
+        _ivars = [ivar for ivar in self.ivars]
+        _gpms = [gpm for gpm in self.gpms]
+        _spec1dfiles = [spec1dfile for spec1dfile in self.spec1dfiles]
+        _objids = [objid for objid in self.objids]
+
+        # good exposures index
+        goodindx_exp = np.arange(self.nexp)
+
+        # check if there are exposures that are completely masked out, i.e., gpms = False for all spectral pixels
+        masked_exps = [np.all(np.logical_not(gpm)) for gpm in _gpms]
+        if np.any(masked_exps):
+            msgs.warn(f'The following exposure(s) is/are completely masked out. It/They will not be coadded.')
+            [msgs.warn(f"Exposure {i}: {fname.split('/')[-1]}  {obj}")
+             for i, (fname, obj, masked_exp) in enumerate(zip(_spec1dfiles, _objids, masked_exps)) if masked_exp]
+            # remove masked out exposure
+            _waves = [wave for (wave, masked_exp) in zip(_waves, masked_exps) if not masked_exp]
+            _fluxes = [flux for (flux, masked_exp) in zip(_fluxes, masked_exps) if not masked_exp]
+            _ivars = [ivar for (ivar, masked_exp) in zip(_ivars, masked_exps) if not masked_exp]
+            _gpms = [gpm for (gpm, masked_exp) in zip(_gpms, masked_exps) if not masked_exp]
+            _spec1dfiles = [spec1dfile for (spec1dfile, masked_exp) in zip(_spec1dfiles, masked_exps) if not masked_exp]
+            _objids = [objid for (objid, masked_exp) in zip(_objids, masked_exps) if not masked_exp]
+            # update good exposures index
+            goodindx_exp = goodindx_exp[np.logical_not(masked_exps)]
+
+        # check if there is still more than 1 exposure left
+        if len(_fluxes) < 2:
+            msgs.error('At least 2 exposures are required for coadding.')
+
+        # check if there is any bad exposure by comparing the rms_sn with the median rms_sn among all exposures
+        if len(_fluxes) > 2:
+            # set sn_smooth_npix if not provided
+            if self.par['sn_smooth_npix'] is None:
+                # same as in coadd.multi_combspec()
+                sn_smooth_npix = int(np.round(0.1 * (np.sum([np.sum(wave > 1.0) for wave in _waves]) / self.nexp)))
```

Comment:
See my feedback on your previous version of this PR. You don't need to assign sn_smooth_pix since that is used for the weights, not for the rms_sn, for which the whole spectrum is used. So all of this sn_smooth_pix should be removed. You should instead write: 

rms_sn, _ = coadd.sn_weights(fluxes, ivars, gpms, const_weights=True)

Reply:
Sorry I missed that. I removed `sn_smooth_pix`.
Thanks!